lib/mcb/revoke_access_wizard.rb

```diff
@@ -7,25 +7,35 @@ def initialize(cli, user, provider)
     end
 
     def run
-      fetch_organisation
+      fetch_organisations
       puts MCB::Render::ActiveRecord.user @user
       confirm_and_remove_user_from_organisation
     end
 
   private
 
-    def fetch_organisation
-      @organisation = @provider.organisations.first # a provider should only ever be associated with one organisation
+    def fetch_organisations
+      @organisations = if @provider
+        # we use first because a provider should only ever be associated with one organisation
+                         [@provider.organisations.first]
+                       else
+                         @user.organisations
+                       end
     end
 
     def confirm_and_remove_user_from_organisation
-      if @user.in?(@organisation.users)
+      if (@user.organisations & @organisations).any?
         puts "\n"
-        puts "You're revoking access for #{@user.email} to organisation #{@organisation.name}."
-        puts MCB::Render::ActiveRecord.providers_table @organisation.providers, name: "Provider access being revoked"
-        @organisation.users.delete(@user) if @cli.agree("Agree?  ")
+        puts "You're revoking access for #{@user.email} to:"
+        puts MCB::Render::ActiveRecord.organisations_table @organisations, name: "Organisation access being revoked"
+        puts MCB::Render::ActiveRecord.providers_table @organisations.map(&:providers).flatten, name: "Provider access being revoked"
```

Comment:
Don't think we need the `.flatten` here.

```suggestion
        puts MCB::Render::ActiveRecord.providers_table @organisations.map(&:providers), name: "Provider access being revoked"
```

Reply:
```
Failures:

  1) mcb users revoke one provider when the user exists and has access to the provider revokes organisation membership to that user
     Failure/Error: table.pack(max_table_width: nil),

     NoMethodError:
       undefined method `id' for #<ActiveRecord::Associations::CollectionProxy []>
       Did you mean?  ids
```